app/controllers/registration/registration.controller.js

```diff
@@ -28,6 +31,39 @@ async function showPasswordPage (req, res, next) {
   }
 }
 
+async function submitPasswordPage (req, res, next) {
+  const sessionData = req.register_invite
+  if (!registrationSessionPresent(sessionData)) {
+    return next(new RegistrationSessionMissingError())
+  }
+
+  const password = req.body['password']
+  const repeatPassword = req.body['repeat-password']
+
+  const errors = {}
+
+  const passwordValidationResult = validatePassword(password)
+  if (!passwordValidationResult.valid) {
+    errors.password = passwordValidationResult.message
+  }
+  if (isEmpty(repeatPassword)) {
+    errors['repeat-password'] = 'Re-type your password'
+  } else if (!errors.password && password !== repeatPassword) {
+    errors['password'] = errors['repeat-password'] = 'Enter same password in both fields'
+  }
```

Comment:
All of this feels like it could be moved out to a separate function (at the least) or a service to help code readability at the controller level and also a little bit easier to test

I don't think this is a blocker.

Would be really good to get around to doing form validation in a uniform way which I know we've talked about a few times internally.

Reply:
After discussion we don't think a half-way implementation of this would be helpful or more readable, this looks like other validation code and we should only significantly differ if we've got guidance on how to do this pattern consistently